### PR TITLE
vendor: resenje.org/singleflight v0.4.3

### PR DESCRIPTION
**Pull request by vvoland**
full diff: https://github.com/janos/singleflight/compare/v0.4.1...v0.4.3

Changes:
- Fix incorrect `Forget` behavior
- Make panic behavior consistent with x/sync package

### CODE REVIEW

**Comment by thaJeztah** [comment]
oh! the diff link doesn't work with the vanity URL, so you need the one from github;

https://github.com/janos/singleflight/compare/v0.4.1...v0.4.3

File: vendor/resenje.org/singleflight/singleflight.go [hunk 1]
```diff
@@ -60,30 +85,45 @@ func (g *Group[K, V]) Do(ctx context.Context, key K, fn func(ctx context.Context
...
+		if !c.forgotten {
+			delete(g.calls, key)
+		}
 	}
 	shared = c.shared
 	g.mu.Unlock()
+
+	if panicErr != nil {
+		panic(panicErr)
+	}
```
Comment (thaJeztah) [inline vendor/resenje.org/singleflight/singleflight.go:125]: Is there any situation where we could trip up a panic now, or would that already be such an exception that it already panicked through other ways, or a panic is the only right thing to do?

Comment (vvoland) [reply vendor/resenje.org/singleflight/singleflight.go:125]: The singleflight behavior for the same key _should_ produce the same result, so it's very likely that the panic would happen again.

The `x/sync/singleflight` also does the same thing: https://cs.opensource.google/go/x/sync/+/refs/tags/v0.9.0:singleflight/singleflight.go;l=102

**Comment by thaJeztah** [comment]
This one is definitely flaky now with runc 1.2.2; I think there were some patches in WIP branches that didn't get merged, and I'm trying to dig them up;

```
=== RUN   TestDiff
    diff_test.go:28: assertion failed: 
        --- expected
        +++ items
          []container.FilesystemChange(
        - 	{{Kind: s"A", Path: "/foo"}, {Kind: s"A", Path: "/foo/bar"}},
        + 	nil,
          )
        
--- FAIL: TestDiff (0.65s)
```

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**